spec/actors/hyku_addons/actors/json_fields_actor_spec.rb

```diff
@@ -0,0 +1,285 @@
+          creator: [
+            [{
+              creator_name_type: "Personal",
+              creator_given_name: "Johnny",
+              creator_family_name: "Testison"
+            }].to_json
+          ]
+        }
+      end
+
+      let(:attributes) do
+        {
+          creator: [
+            { "creator_name_type" => "Personal", "creator_given_name" => "Liang", "creator_family_name" => "Gao" },
+            { "creator_name_type" => "Personal", "creator_given_name" => "Lidai", "creator_family_name" => "Wang" }
+          ],
+          editor: [
+            { "editor_name_type" => "Personal", "editor_given_name" => "Alexander A.", "editor_family_name" => "Oraevsky" },
+            { "editor_name_type" => "Personal", "editor_given_name" => "Lihong V.", "editor_family_name" => "Wang" }
+          ]
+        }
+      end
+
+      let(:transformed_attributes) do
+        { "creator" => ["[{\"creator_name_type\":\"Personal\",\"creator_given_name\":\"Liang\",\"creator_family_name\":\"Gao\"},{\"creator_name_type\":\"Personal\",\"creator_given_name\":\"Lidai\",\"creator_family_name\":\"Wang\"}]"], "editor" => ["[{\"editor_name_type\":\"Personal\",\"editor_given_name\":\"Alexander A.\",\"editor_family_name\":\"Oraevsky\"},{\"editor_name_type\":\"Personal\",\"editor_given_name\":\"Lihong V.\",\"editor_family_name\":\"Wang\"}]"], "contributor" => [], "funder" => [], "alternate_identifier" => [], "related_identifier" => [], "current_he_institution" => [] }
+      end
+
+      it "adds them all" do
+        expect(env.attributes).to eq(transformed_attributes)
+      end
+    end
+
+    context "with blank or empty attributes" do
+      let(:original_attributes) do
+        {
+          creator: [
+            [{
+              creator_name_type: "Personal",
+              creator_given_name: "Johnny",
+              creator_family_name: "Testison"
+            }].to_json
+          ],
+          editor: [
+            [{
+              creator_name_type: "Personal",
+              creator_given_name: "Johnny",
+              creator_family_name: "Testison"
+            }].to_json
+          ]
+        }
+      end
+      let(:attributes) do
+        {
+          creator: [
+            {}
+          ],
+          editor: [
+            { "editor_name_type" => "Personal", "editor_given_name" => "Alexander A.", "editor_family_name" => nil }
+          ]
+        }
+      end
+
+      let(:transformed_attributes) do
+        { "editor" => ["[{\"editor_name_type\":\"Personal\",\"editor_given_name\":\"Alexander A.\",\"editor_family_name\":null}]"], "creator" => [], "contributor" => [], "funder" => [], "alternate_identifier" => [], "related_identifier" => [], "current_he_institution" => [] }
+      end
+
+      it "turns fields into null or adds empty arrays" do
+        expect(env.attributes).to eq(transformed_attributes)
+      end
+    end
+
+    context "with no name type" do
+      let(:original_attributes) do
+        {
+          editor: [
```

Comment:
Here the hash key is **editor** should the sub keys not be prefixed with editor instead of creator rg
  
      editor_name_type

as against

    creator_name_type